python/ambassador/envoy/v3/v3listener.py

```diff
@@ -582,17 +584,9 @@ def finalize_tcp(self) -> None:
 
                 # If we have a context...
                 if chain.context:
-                    # ...then we can do TLS...
+                    # ...then we can ask for TLS.
                     filter_chain_match["transport_protocol"] = "tls"
 
-                    # ...and match server names.
-                    #
-                    # (The if here is because if chain_hosts is empty, it means '*'. If there are
-                    # no specific names in the list, or if one of the names is actually '*', don't 
-                    # bother matching: just accept everything.)
-                    if (len(chain_hosts) > 0) and ('*' not in chain_hosts):
-                        filter_chain_match['server_names'] = chain_hosts
-
```

Comment:
Don't panic: this is a move, not a deletion. We handle `server_names` below now.